litmus-portal/cluster-agents/subscriber/pkg/cluster/events/workflow.go

```diff
@@ -68,6 +69,9 @@ func startWatch(stopCh <-chan struct{}, s cache.SharedIndexInformer, stream chan
 // responsible for extracting the required data from the event and streaming
 func workflowEventHandler(obj interface{}, eventType string, stream chan types.WorkflowEvent, startTime int64) {
 	workflowObj := obj.(*v1alpha1.Workflow)
+	if val, ok := workflowObj.Labels["workflows.argoproj.io/controller-instanceid"]; !ok || val != ClusterID {
+		return
```

Comment:
A log might be helpful here?

Reply:
I thought it might clutter the logs that's why I omitted it, a single workflow run can generate multiple events and the log would happen for each of them. Should i still add it?

Reply:
Have update the PR